File: service/src/main/java/com/sethy/service/comment/service/CommentService.java
package com.sethy.service.comment.service;

import com.sethy.service.comment.dto.CommentResponse;
import com.sethy.service.comment.dto.CreateCommentRequest;
import com.sethy.service.comment.dto.UpdateCommentRequest;

import java.util.List;

public interface CommentService {

    CommentResponse createComment(Long ticketId, CreateCommentRequest request, String username);

    List<CommentResponse> getCommentsForTicket(Long ticketId, String username);

    CommentResponse updateComment(Long ticketId, Long commentId, UpdateCommentRequest request, String username);

    void deleteComment(Long ticketId, Long commentId, String username);
}

File: web/src/services/comment.service.ts
import type { Comment, CreateCommentRequest, UpdateCommentRequest } from "@/types/comment";
import { api } from "@/utils/api";

class CommentService {
  async getCommentsForTicket(ticketId: number) {
    return await api.get<Comment[]>(`/tickets/${ticketId}/comments`);
  }

  async createComment(ticketId: number, request: CreateCommentRequest) {
    return await api.post<Comment>(`/tickets/${ticketId}/comments`, request);
  }

  async updateComment(ticketId: number, commentId: number, request: UpdateCommentRequest) {
    return await api.put<Comment>(`/tickets/${ticketId}/comments/${commentId}`, request);
  }

  async deleteComment(ticketId: number, commentId: number) {
    await api.delete<void>(`/tickets/${ticketId}/comments/${commentId}`);
  }
}

export const commentService = new CommentService();
